### PR TITLE
Obtain correct Rx lenght during OOB transaction.

### PR DESCRIPTION
**Pull request by albertofloyd**
drivers: espi: Obtain correct Rx lenght during OOB transaction.

MEC15xx eSPI OOB RX length register holds received message length and receive buffer length, need to extract rcvd msg length.

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

### CODE REVIEW

File: drivers/espi/espi_mchp_xec.c [hunk 1]
```diff
@@ -417,24 +417,25 @@ static int espi_xec_send_oob(struct device *dev, struct espi_oob_packet pckt)
...
+		LOG_ERR("insufficient space");
 		return -EINVAL;
 	}
 
 	memcpy(slave_tx_mem, pckt.buf, pckt.len);
 
 	ESPI_OOB_REGS->TX_LEN = pckt.len;
 	ESPI_OOB_REGS->TX_CTRL = MCHP_ESPI_OOB_TX_CTRL_START;
-	LOG_DBG("%s %d\n", __func__, ESPI_OOB_REGS->TX_LEN);
+	LOG_ERR("%s %d", __func__, ESPI_OOB_REGS->TX_LEN);
```
Comment (dcpleung) [inline drivers/espi/espi_mchp_xec.c]: I think this should be `LOG_DBG()` instead.

File: drivers/espi/espi_mchp_xec.c [hunk 2]
```diff
@@ -471,11 +469,13 @@ static int espi_xec_receive_oob(struct device *dev,
 	}
 
 	/* Check if buffer passed to driver can fit the received packet */
-	if (ESPI_OOB_REGS->RX_LEN > pckt.len) {
+	if ((ESPI_OOB_REGS->RX_LEN & ESPI_XEC_OOB_RX_LEN_MASK) > pckt.len) {
+		LOG_ERR("space rcvd %d vs %d",
+			ESPI_OOB_REGS->RX_LEN, pckt.len);
```
Comment (dcpleung) [inline drivers/espi/espi_mchp_xec.c]: Should this `RX_LEN` need to be masked too? Also please expand the log message to explain the error (maybe something like "received oob packet longer than buffer"?).

Comment (albertofloyd) [reply drivers/espi/espi_mchp_xec.c]: Yes

File: drivers/espi/espi_mchp_xec.c (hunk 1 above)
Comment (albertofloyd) [reply drivers/espi/espi_mchp_xec.c]: fixed

**Review by franciscomunoz — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]

**Pull request merged**